assertThrownBy misreports chained assertion failures as type-check errors (sticky describedAs + overridingErrorMessage)

## Description

`assertThrownBy` returns a `ThrowableAssert` that carries a sticky `describedAs` description and `overridingErrorMessage` override. Both were intended for the "wrong type / nothing thrown" diagnostics inside `assertThrownBy` itself, but because AssertJ applies them to the assert object's `WritableAssertionInfo`, they stick to **every assertion chained afterwards** on the returned object. A later message assertion that fails is therefore reported as an instance-of (type-check) failure with the original `assertThrownBy` description — extremely misleading, because the type check actually passed.

Root cause — `fluent-assert-core` `me/ahoo/test/asserts/Throwable.kt:49-55`:

```kotlin
val throwable = Assertions.catchThrowable(shouldRaiseThrowable)
return throwable
    .assert()
    .describedAs { "Expected ${throwableType.simpleName} to be thrown, but was: $throwable" }
    .isNotNull()
    .overridingErrorMessage(shouldBeInstance(throwable, throwableType).create())
    .isInstanceOf(throwableType) as ThrowableAssert<T>
```

`overridingErrorMessage(...)` replaces the failure message of all subsequent assertions on that assert instance, and `describedAs(...)` likewise persists.

## Steps to Reproduce

```kotlin
import me.ahoo.test.asserts.assertThrownBy
import org.junit.jupiter.api.Test

class DemoTest {
    @Test
    fun chainedMessageAssertionFailsAsTypeError() {
        assertThrownBy<IllegalArgumentException> {
            throw IllegalArgumentException("boom-expected")
        }.hasMessageContaining("wrong-fragment") // type matches; only the fragment is wrong
    }
}
```

## Expected Behavior

A message-contains diagnostic pointing at the fragment, e.g. `Expecting actual: "boom-expected" to contain: "wrong-fragment"`. The exception type check passed and should not appear as the failure.

## Actual Behavior

```
java.lang.AssertionError: [Expected IllegalArgumentException to be thrown, but was: java.lang.IllegalArgumentException: boom-expected]
Expecting actual throwable to be an instance of:
  java.lang.IllegalArgumentException
but was:
  java.lang.IllegalArgumentException: boom-expected
```

The failure reads as a type mismatch ("Expected IllegalArgumentException ... but was: IllegalArgumentException"), which sends users to debug a phantom type/classloader problem instead of their message fragment. Confirmed empirically: the same chained form **passes** when both the type and the message fragment match, so only the diagnostic is wrong.

## Suggested Fix

Keep the rich diagnostics for `assertThrownBy`'s own failures (nothing thrown / wrong type) but return a clean assert for chaining, e.g.:

```kotlin
fun <T : Throwable> assertThrownBy(throwableType: Class<T>, shouldRaiseThrowable: () -> Unit): ThrowableAssert<T> {
    val throwable = Assertions.catchThrowable(shouldRaiseThrowable)
    if (throwable == null || !throwableType.isInstance(throwable)) {
        throw AssertionError(
            "Expected ${throwableType.name} to be thrown, but was: $throwable",
            shouldBeInstance(throwable ?: Any(), throwableType).create()?.let(::AssertionError)
        )
    }
    return throwable.assert() as ThrowableAssert<T>
}
```

(or throw `shouldBeInstance(...).create()`-based failures directly instead of routing them through the returned assert). The KDoc already demonstrates chained usage, so making chained diagnostics accurate also matches the documented API contract.

## Environment

- fluent-assert-core 1.0.0 (pattern also present in 0.2.8)
- AssertJ 3.27.7, Kotlin 2.4.10
- Discovered while testing CoApi (me.ahoo.coapi 2.2.0); documented in that repo's testing guidance as a workaround (try/catch pattern) in https://github.com/Ahoo-Wang/CoApi/blob/main/skills/coapi-developer/references/testing.md

--- a/core/src/main/kotlin/me/ahoo/test/asserts/Throwable.kt
+++ b/core/src/main/kotlin/me/ahoo/test/asserts/Throwable.kt
@@ -60,12 +60,16 @@ fun <T : Throwable> assertThrownBy(
     shouldRaiseThrowable: () -> Unit
 ): ThrowableAssert<T> {
     val throwable = Assertions.catchThrowable(shouldRaiseThrowable)
-    return throwable
+    // describedAs/overridingErrorMessage persist on the assert instance's WritableAssertionInfo,
+    // so run the diagnostics on a throwaway instance and return a fresh one — otherwise chained
+    // assertions report their failures as type-check errors.
+    throwable
         .assert()
         .describedAs { "Expected ${throwableType.simpleName} to be thrown, but was: $throwable" }
         .isNotNull()
         .overridingErrorMessage(shouldBeInstance(throwable, throwableType).create())
-        .isInstanceOf(throwableType) as ThrowableAssert<T>
+        .isInstanceOf(throwableType)
+    return throwable.assert() as ThrowableAssert<T>
 }
 
 /**

--- a/core/src/test/kotlin/me/ahoo/test/asserts/ThrowableTest.kt
+++ b/core/src/test/kotlin/me/ahoo/test/asserts/ThrowableTest.kt
@@ -2,6 +2,7 @@ package me.ahoo.test.asserts
 
 import org.assertj.core.api.ThrowableAssert
 import org.junit.jupiter.api.Assertions.assertInstanceOf
+import org.junit.jupiter.api.Assertions.assertThrows
 import org.junit.jupiter.api.Test
 
 class ThrowableTest {
@@ -37,5 +38,25 @@ class ThrowableTest {
                 throw IllegalStateException("wrong")
             }
         }.hasMessageContaining(IllegalArgumentException::class.java.name)
+    }
+
+    @Test
+    fun `given matching type when chained message assertion fails then failure reports message not type`() {
+        val error = assertThrows(AssertionError::class.java) {
+            assertThrownBy<IllegalArgumentException> {
+                throw IllegalArgumentException("boom-expected")
+            }.hasMessageContaining("wrong-fragment")
+        }
+        error.message!!.assert()
+            .contains("wrong-fragment")
+            .contains("boom-expected")
+        error.message!!.assert().doesNotContain("to be an instance of")
+    }
+
+    @Test
+    fun `given matching type and message when assertThrownBy then chained assertions pass`() {
+        assertThrownBy<IllegalArgumentException> {
+            throw IllegalArgumentException("boom-expected")
+        }.hasMessage("boom-expected").hasMessageContaining("boom")
     }
 }

